### PR TITLE
hfrunner.classify should return list[list[float]] not list[str]

### PR DESCRIPTION
**Pull request by nwaughachukwuma**
## Purpose
Fix the return type of HfRunner.classify - should be list[list[float]] and not list[str]

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -459,14 +459,33 @@ def get_prompt_embeddings(self, prompts: list[str]) -> list[torch.Tensor]:
...
+    def classify(self, prompts: list[str]) -> list[list[float]]:
+        from sentence_transformers import SentenceTransformer, CrossEncoder
+
+        # Prevent misuse with incompatible model types
+        if isinstance(self.model, (SentenceTransformer, CrossEncoder)):
+            raise TypeError(
+                "classify() is only supported for standard Hugging Face classification models "
+                "(e.g., AutoModelForSequenceClassification). "
+                "SentenceTransformer and CrossEncoder are not supported."
+            )
```
Comment (noooop) [inline tests/conftest.py]: Thanks for your contribution.

The tests/conftest.py is only used by developers, so we should trust that they know what they're doing. There's no need for such excessive safeguards.

Please make DCO happy.

Comment (nwaughachukwuma) [reply tests/conftest.py]: Thanks for the feedback, @noooop.

I've removed that code block. Should be fine now.

File: tests/conftest.py [hunk 2]
```diff
@@ -459,14 +459,23 @@ def get_prompt_embeddings(self, prompts: list[str]) -> list[torch.Tensor]:
...
             output = self.model(**self.wrap_device(inputs))
+            # NOTE: In practice, HF models return ModelOutput with a `logits` attribute (torch.Tensor)
+            if not hasattr(output, "logits"):
+                raise TypeError("Model output does not have a 'logits' attribute.")
+
+            if not isinstance(output.logits, torch.Tensor):
+                raise TypeError(
+                    f"'logits' is {type(output.logits)}, expected torch.Tensor"
+                )
+
```
Comment (noooop) [inline tests/conftest.py]: This check may not be necessary, as developers would ultimately need to debug the issue if the ModelOutput is missing a logits attribute.

Comment (nwaughachukwuma) [reply tests/conftest.py]: Ok, I kept it there for type safety. Given developers would ultimately debug if there's an issue, I've replaced it with an assert statement. I hope it works better.

```py
# Before
if not hasattr(output, "logits"):
   raise TypeError("Model output does not have a 'logits' attribute.")

if not isinstance(output.logits, torch.Tensor):
   raise TypeError(f"'logits' is {type(output.logits)}, expected torch.Tensor")

# After
assert isinstance(output.logits, torch.Tensor)
```
@noooop

**Review by noooop — APPROVED** [review]
